File: air01.js
// fonctions utilisées

function custom_include(array, element)
{
	for (let i = 0; i < array.length; i++)
	{
		if (array[i] === element)
			return true;
	}
	return false;
}

function ma_fonction(str, string_séparateur)
{
	let word = [];
	let tableau = [];
	let tmp = []
	let i = 0;
	let j = 0;
	let k = 0;

	
	while(i <= str.length)
	{
		// if (str.charAt(i) !== string_séparateur.charAt(1) || str.charAt(i - 1) !== string_séparateur.charAt(0))
		if (str.charAt(i) !== string_séparateur.charAt(0) || str.charAt(i + 1) !== string_séparateur.charAt(1))
		{
			if (i == str.length)
			{
				tmp.push(String(word.join('')));
				tableau.push(tmp);
			}
			// console.log("Youpi")
			word.push(str[i]);
		}
		else
		{

			for (j = 0; j < string_séparateur.length; j++)
			{
				// if (str[i + j] != string_séparateur[j])
				if (str.charAt(i + j) !== string_séparateur.charAt(j))
				{
					k = i + j - 1
					// for (; j > 0; j--)
					// for (let k = 0; k < j; k++)
					// while(i < k)
					// {
						word.push(str[i]);
						console.log(`down ${k}`);
					// 	// i++;
						// i++;
					// }
					// i += 2;
					break;
				}
				// else
				// {
				// 	console.log(`${j}`);
				// }
			}
			tmp.push(String(word.join('')));
			tableau.push(tmp);
			tmp = [];
			word = [];
			i += string_séparateur.length - 1
		}
		i++;
	}
	return (tableau)
}

function is_arg_valid()
{
	if (is_nb_arg_correct === false)
		return false;

	if (process.argv[3].length < 2)
		return false;
}

// Partie 1 : Gestion d'erreur
is_nb_arg_correct = process.argv.length === 4;

if (is_arg_valid() === false)
{
	console.log("error");
	process.exit(1);
}

// Partie 2 : Parsing
const arg = process.argv[2];
const separator = process.argv[3];
console.log(`max i = ${arg.length - 1}`)

// Partie 3 : Résolution
const answer = ma_fonction(arg, separator);

// Partie 4 : Affichage
for (let i in answer)
	console.log(answer[i])

File: air01.py
import sys

ascii_space = 32
ascii_horizontal_tab = 10
ascii_line = 9

# fonctions utilisées

def is_white_space(char):
  if ord(char) == ascii_space or ord(char) == ascii_line or ord(char) == ascii_horizontal_tab:
    return True

def ma_fonction(string_à_couper, string_séparateur):
  word = []
  tableau = []
  tmp = []
  comparaison = " " + str(string_séparateur[0 : len(string_séparateur)]) + " "
  i = 0
  while i < len(string_à_couper):
   if string_à_couper[i - 1 : i + len(string_séparateur) + 1] == comparaison:
      tmp.append(''.join(word))
      tableau.append(tmp)
      tmp = []
      word = []
      i += len(string_séparateur)
   else:
      word.append(string_à_couper[i])
   i += 1
  tmp.append(''.join(word))
  tableau.append(tmp)
  return tableau

def is_arg_valid():
  if is_nb_arg_correct == False:
    return False

# Partie 1 : Gestion d'erreur
is_nb_arg_correct = len(sys.argv) == 3

if is_arg_valid() == False:
  sys.exit("error")

# Partie 2 : Parsing
arg = sys.argv[1]
separator = sys.argv[2]

# Partie 3 : Résolution
split = ma_fonction(arg, separator)

# Partie 4 : Affichage
for word in split:
   print(word)
